### PR TITLE
KOGITO-2416 - Review KogitoEmptyState component for improved reusability

### PR DESCRIPTION
**Pull request by cristianonicolai**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@AjayJagan I tried to decouple the empty state from the functions that are needed to execute once the button is clicked. Also doesn't seem that running _onFIlterClick_ as an async function is necessary, see https://github.com/kiegroup/kogito-apps/pull/259/files#diff-1818f66c4b3f8e8abf534ef55f6981f7R85 or am I missing something?

**Comment by cristianonicolai** [comment]
@jstastny-cz we might need to find another way to wrap ouia, removing the default export makes a lot easier to handle and it seems to make harder for when you need to export multiple types, see https://github.com/kiegroup/kogito-apps/pull/259/files#diff-9395e6e344be30633b1723848e8871aaR32

**Comment by cristianonicolai** [comment]
will leave this PR open until https://issues.redhat.com/browse/KOGITO-2386 is fixed and test is properly executed in the common module.

**Comment by AjayJagan** [comment]
> @AjayJagan I tried to decouple the empty state from the functions that are needed to execute once the button is clicked. Also doesn't seem that running _onFIlterClick_ as an async function is necessary, see https://github.com/kiegroup/kogito-apps/pull/259/files#diff-1818f66c4b3f8e8abf534ef55f6981f7R85 or am I missing something?

@cristianonicolai  - yep async can be removed(my bad .. sorry)

**Comment by jstastny-cz** [comment]
> @jstastny-cz we might need to find another way to wrap ouia, removing the default export makes a lot easier to handle and it seems to make harder for when you need to export multiple types, see https://github.com/kiegroup/kogito-apps/pull/259/files#diff-9395e6e344be30633b1723848e8871aaR32

Ok I can rewrite it after using the 'rename export':
```
const Component:React.FC<> = ({}){
.
.
}
const ComponentWithOuia = withOuiaContext(Component)
export {ComponentWithOuia as Component}
```

This is actually how it's implemented in patternfly version we use. When you look in master, there's no such thing ... The reason for this wrapper was IMO to enable turn on/off of the ouia support, while in latest it's on by default....

**Comment by cristianonicolai** [comment]
@AjayJagan @jstastny-cz ready for review.

File: packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
 
-exports[`Emptystate component tests snapshot testing 1`] = `
+exports[`KogitoEmptyState component tests Info test 1`] = `
+<Bullseye>
+  <EmptyState
+    variant="full"
+  >
+    <EmptyStateIcon
+      color="var(--pf-global--info-color--100)"
+      icon={[Function]}
```
Comment (jstastny-cz) [inline packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: the differentiation based on types is not visible in these shallow snapshots. Is it possible to use mount instead? Maybe we can even mock the EmptyStateIcon, but using mount we should see the actual icon passed to the component.

**Review by jstastny-cz — COMMENTED** [review]
Just one question about snapshots, looks good otherwise.

File: packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap (hunk 1 above)
Comment (cristianonicolai) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: yeah, makes sense. So now I changed to `mount` but I couldn't figure how to mock only the `EmptyStateIcon`. I could only mock the entire `react-core` module. @AjayJagan any tips on how to do that?
This fails with finding the module.
`jest.mock('@patternfly/react-core/EmptyState/EmptyStateIcon');`
And this ends up mocking all components, then it stops at the `Bullseye` component which is not what we want.
`jest.mock('@patternfly/react-core')`

Comment (jstastny-cz) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: You can do this to mock just a single component from whole package:
```
const MockedComponent = (): React.ReactElement => {
  return <></>;
};

jest.mock('@patternfly/react-core', () => ({
  ...jest.requireActual('@patternfly/react-core'),
  EmptyStateIcon: () => {
    return <MockedComponent />;
  }
}));
```

Comment (cristianonicolai) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: mock works but then I get the following in the snapshot:

```
<EmptyStateIcon
            color="var(--pf-global--warning-color--100)"
            icon={[Function]}
            size="sm"
          >
        <EmptyStateIconMock />
</EmptyStateIcon>
```

Comment (jstastny-cz) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: ah, sorry, I didn't realize, so we can either keep it full or just mock the icons passed to EmptyStateIcon. That would mean mocking ''@patternfly/react-icons'' instead:

```
const MockedComponent = (): React.ReactElement => {
  return <></>;
};

jest.mock('@patternfly/react-icons', () => ({
  ...jest.requireActual('@patternfly/react-icons'),
  InfoCircleIcon: () => {
    return <MockedComponent />;
  },
  ExclamationTriangleIcon: () => { return <MockedComponent /> },
.
.
.
}));
```

Comment (jstastny-cz) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: We'd still save ~80 lines out of 380 in the snapshots. And reducing the noise.

Comment (cristianonicolai) [reply packages/common/src/components/Atoms/KogitoEmptyState/tests/__snapshots__/KogitoEmptyState.test.tsx.snap]: @jstastny-cz thanks that helped. Have a look now

**Review by AjayJagan — APPROVED** [review]
working fine :+1:

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**